diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/classes/SingleConstructorRule.kt

```diff
@@ -0,0 +1,151 @@
+package org.cqfn.diktat.ruleset.rules.classes
+
+import com.pinterest.ktlint.core.Rule
+import com.pinterest.ktlint.core.ast.ElementType
+import com.pinterest.ktlint.core.ast.ElementType.CLASS
+import com.pinterest.ktlint.core.ast.ElementType.CLASS_BODY
+import com.pinterest.ktlint.core.ast.ElementType.LBRACE
+import com.pinterest.ktlint.core.ast.ElementType.PRIMARY_CONSTRUCTOR
+import com.pinterest.ktlint.core.ast.ElementType.SECONDARY_CONSTRUCTOR
+import com.pinterest.ktlint.core.ast.ElementType.WHITE_SPACE
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings.SINGLE_CONSTRUCTOR_SHOULD_BE_PRIMARY
+import org.cqfn.diktat.ruleset.utils.KotlinParser
+import org.cqfn.diktat.ruleset.utils.findAllNodesWithSpecificType
+import org.cqfn.diktat.ruleset.utils.getAllChildrenWithType
+import org.cqfn.diktat.ruleset.utils.getIdentifierName
+import org.cqfn.diktat.ruleset.utils.isGoingAfter
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.PsiWhiteSpaceImpl
+import org.jetbrains.kotlin.psi.KtBinaryExpression
+import org.jetbrains.kotlin.psi.KtClass
+import org.jetbrains.kotlin.psi.KtDotQualifiedExpression
+import org.jetbrains.kotlin.psi.KtExpression
+import org.jetbrains.kotlin.psi.KtNameReferenceExpression
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.psi.KtSecondaryConstructor
+import org.jetbrains.kotlin.psi.KtThisExpression
+import org.jetbrains.kotlin.psi.psiUtil.asAssignment
+
+class SingleConstructorRule(private val config: List<RulesConfig>) : Rule("single-constructor") {
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+    private val kotlinParser by lazy { KotlinParser() }
+
+    override fun visit(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit
+    ) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        if (node.elementType == CLASS) {
+            handleClassConstructors(node)
+        }
+    }
+
+    private fun handleClassConstructors(node: ASTNode) {
+        if (node.findChildByType(PRIMARY_CONSTRUCTOR) == null) {
+            // class has no primary constructor, need to count secondary constructors
+            node
+                .findChildByType(CLASS_BODY)
+                ?.getAllChildrenWithType(SECONDARY_CONSTRUCTOR)
+                ?.singleOrNull()
+                ?.let { secondaryCtor ->
+                    SINGLE_CONSTRUCTOR_SHOULD_BE_PRIMARY.warnAndFix(
+                        config, emitWarn, isFixMode, "in class <${node.getIdentifierName()?.text}>",
+                        node.startOffset, node) {
+                        convertConstructorToPrimary(node, secondaryCtor)
```

Comment:
There are no spaces (indents) here or it is github showing like that? 
Is it how our code style works?

Reply:
There is indented argument list and then code block with same indent. This is how idea works; our code style does this too, with optional extended indent of argument list.